tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -0,0 +1,236 @@
+"""Delete and re-parse a set of datafiles."""
+
+from django.core.management.base import BaseCommand
+from django.core.management import call_command
+from elasticsearch.helpers.errors import BulkIndexError
+from tdpservice.data_files.models import DataFile
+from tdpservice.scheduling import parser_task
+from tdpservice.search_indexes.documents import tanf, ssp, tribal
+from tdpservice.core.utils import log
+from django.contrib.admin.models import ADDITION
+from tdpservice.users.models import User
+from datetime import datetime
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Command class."""
+
+    help = "Delete and re-parse a set of datafiles. All re-parsed data will be moved into a new set of Elastic indexes."
+
+    def add_arguments(self, parser):
+        """Add arguments to the management command."""
+        parser.add_argument("-q", "--fiscal_quarter", type=str, help="Re-parse all files in the fiscal quarter, "
+                            "e.g. Q1.")
+        parser.add_argument("-y", "--fiscal_year", type=str, help="Re-parse all files in the fiscal year, e.g. 2021.")
+        parser.add_argument("-a", "--all", action='store_true', help="Clean and re-parse all datafiles. If selected, "
+                            "fiscal_year/quarter aren't necessary.")
+        parser.add_argument("-n", "--new_indices", action='store_true', help="Move re-parsed data to new Elastic "
+                            "indices.")
+        parser.add_argument("-d", "--delete_indices", action='store_true', help="Requires new_indices. Delete the "
+                            "current Elastic indices.")
+
+    def __get_log_context(self, system_user):
+        """Return logger context."""
+        context = {'user_id': system_user.id,
+                   'action_flag': ADDITION,
+                   'object_repr': "Clean and Re-parse"
+                   }
+        return context
+
+    def __backup(self, backup_file_name, log_context):
+        """Execute Postgres DB backup."""
+        try:
+            logger.info("Beginning re-parse DB Backup.")
+            call_command('backup_db', '-b', '-f', f'{backup_file_name}')
+            logger.info("Backup complete! Commencing clean and re-parse.")
+
+            log("Database backup complete.",
+                logger_context=log_context,
+                level='info')
+        except Exception as e:
+            log("Database backup FAILED. Clean and re-parse NOT executed. Database and Elastic are CONSISTENT!",
+                logger_context=log_context,
+                level='error')
+            raise e
```

Comment:
You can just log the error and then return

Reply:
I personally think it is very important that this command raises an exception in any situation that it runs into a failure to be extremely explicit about the fact that it failed and the database and elastic could be very inconsistent. I know that isn't the case with this exception, but it sets the precedence for consistency across the command. Maybe @andrew-jameson  or @jtimpe could provide some extra feedback?

Reply:
I think I agree with @elipe17 that we should raise whatever the exception is here.